src/store/menu/menuSlice.ts

```diff
@@ -0,0 +1,32 @@
+import { createSlice } from '@reduxjs/toolkit';
+import type { RootState } from '../store';
+
+const initialState = {
+  isProfileMenuOpen: false,
+  isMobileMenuOpen: false,
+};
+
+export const menuSlice = createSlice({
+  name: 'menu',
+  initialState,
+  reducers: {
+    handleMobileMenuChange: (state) => {
+      const stateValue = { ...state };
```

Comment:
Don't do it in redux toolkit.

You will probably make an additional redundant spreading what could be very tough for performance.

We use toolkit, so it will take care of these stuff

Reply:
add correction